app/assets/javascripts/components/timeline/timeline_handler.jsx

```diff
@@ -19,79 +18,65 @@ import {
-    children: PropTypes.node,
-    controls: PropTypes.func,
-    weeks: PropTypes.array.isRequired,
-    loading: PropTypes.bool,
-    editableBlockIds: PropTypes.array,
-    all_training_modules: PropTypes.array,
-    fetchAllTrainingModules: PropTypes.func.isRequired,
-    editPermissions: PropTypes.bool.isRequired
-  },
-
-  getInitialState() {
-    return {
-      reorderable: false,
-      editableTitles: false
-    };
-  },
-
-  componentDidMount() {
-    document.title = `${this.props.course.title} - ${I18n.t('courses.timeline_link')}`;
-    return this.props.fetchAllTrainingModules();
-  },
-
-  _cancelBlockEditable(blockId) {
+// Define TimelineHandler as a functional component using an arrow function
+// Move propTypes outside the component definition
+// Replace state variables with useState hooks
+const TimelineHandler = (props) => {
+  const [reorderable, setReorderable] = useState(false);
+  const [editableTitles, setEditableTitles] = useState(false);
+
+// Replace componentDidMount with useEffect hook
+  useEffect(() => {
+    document.title = `${props.course.title} - ${I18n.t('courses.timeline_link')}`;
+    props.fetchAllTrainingModules();
+  }, [props.course.title, props.fetchAllTrainingModules]);
+
+// Convert class methods to regular functions within the component
+  const _cancelBlockEditable = (blockId) => {
     // TODO: Restore to persisted state for this block only
-    return this.props.cancelBlockEditable(blockId);
-  },
+    props.cancelBlockEditable(blockId);
+  };
 
-  _cancelGlobalChanges() {
-    this.setState({ reorderable: false });
-    this.setState({ editableTitles: false });
-    this.props.restoreTimeline();
-  },
+  const _cancelGlobalChanges = () => {
+    setReorderable(false);
+    setEditableTitles(false);
+    props.restoreTimeline();
+  };
 
-  _enableReorderable() {
-    return this.setState({ reorderable: true });
-  },
+  const _enableReorderable = () => {
+    setReorderable(true);
+  };
 
-  _enableEditTitles() {
-    return this.setState({ editableTitles: true });
-  },
+  const _enableEditTitles = () => {
+    setEditableTitles(true);
+  };
 
-  _resetTitles() {
+  const _resetTitles = () => {
     if (confirm(I18n.t('timeline.reset_titles_confirmation'))) {
-      this.props.resetTitles();
-      this.saveTimeline();
+      props.resetTitles();
+      saveTimeline();
     }
-  },
-
-  saveTimeline() {
-    this.setState({ reorderable: false });
-    this.setState({ editableTitles: false });
-    const toSave = { weeks: this.props.weeks };
-    this.props.persistTimeline(toSave, this.props.course_id);
-  },
-
-  render() {
-    if (this.props.course.loading) {
+  };
+
+  const saveTimeline = () => {
+    setReorderable(false);
+    setEditableTitles(false);
+    const toSave = { weeks: props.weeks };
+    props.persistTimeline(toSave, props.course_id);
+  };
+
+// Replace render method with a separate renderTimelineHandler function
+  const renderTimelineHandler = () => {
```

Comment:
Why implement this as a function instead of just returning a JSX node from main functional component?

Reply:
yaa, we don't really need a separate function here. I think to simplify this by moving the logic directly into the main component's return statement

Reply:
> Why implement this as a function instead of just returning a JSX node from main functional component?

I had done that change , you can preview it